manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java

```diff
@@ -73,41 +75,47 @@ public Bridge createDependencies(Work work, Bridge bridge) {
         rhoasService.createTopicAndGrantAccessFor(resourceNamesProvider.getBridgeErrorTopicName(bridge.getId()),
                 RhoasTopicAccessType.CONSUMER_AND_PRODUCER);
 
-        // We don't need to wait for the Bridge to be READY to create the Error Handler.
-        createErrorHandlerProcessor(bridge);
+        // We don't need to wait for the Bridge to be READY to handle the Error Handler.
+        createOrUpdateOrDeleteErrorHandlerProcessor(bridge);
 
         bridge.setDependencyStatus(ManagedResourceStatus.READY);
         return persist(bridge);
     }
 
     /**
-     * Creates error handler processor if required
+     * Creates, updates or deletes Error Handler processor as required
      *
      * @param bridge input bridge
-     * @return true if the work can proceed (either the error handler processor
-     *         is not required or it's created and ready), false otherwise.
      */
-    private void createErrorHandlerProcessor(Bridge bridge) {
-        // If an ErrorHandler is not needed, consider it ready
+    private void createOrUpdateOrDeleteErrorHandlerProcessor(Bridge bridge) {
+        String bridgeId = bridge.getId();
+        String customerId = bridge.getCustomerId();
+        ListResult<Processor> hiddenProcessors = processorService.getHiddenProcessors(bridgeId, customerId);
+
+        // If an ErrorHandler is not defined, consider it ready and delete any lingering instances
         Action errorHandlerAction = bridge.getDefinition().getErrorHandler();
         boolean errorHandlerProcessorIsNotRequired = Objects.isNull(errorHandlerAction);
         if (errorHandlerProcessorIsNotRequired) {
+            deleteErrorHandlingProcessor(bridge);
             return;
         }
 
-        String bridgeId = bridge.getId();
-        String customerId = bridge.getCustomerId();
-        ListResult<Processor> processors = processorService.getHiddenProcessors(bridgeId, customerId);
-
+        // If an Error Handler processor exists assume it is to be updated otherwise create it!
         // This assumes we can only have one ErrorHandler Processor per Bridge
-        if (processors.getTotal() > 0) {
-            return;
+        if (hiddenProcessors.getTotal() > 0) {
+            // Update Error Handler processor if already present
+            Processor errorHandler = hiddenProcessors.getItems().get(0);
```

Comment:
a nitpick - maybe throw an exception in case there are several hidden Processors as it is not possible to identify which one is error handler?

Reply:
I have refactored slightly to handle the first Error Handler processor found.